src/main/scala/is/hail/HailContext.scala

```diff
@@ -164,7 +164,8 @@ object HailContext {
     ProgressBarBuilder.build(sparkContext)
 
     val sqlContext = new org.apache.spark.sql.SQLContext(sparkContext)
-    val hc = new HailContext(sparkContext, sqlContext, tmpDir, branchingFactor)
+    val hcTemp = TempDir.createTempDir(tmpDir, sparkContext.hadoopConfiguration)
```

Comment:
Why do you need to create the temp directory? Was it not working before?

Reply:
This creates a directory within the provided directory that should get (recursively) deleted when the session exits.

Reply:
We're starting to have cases where we want to use these to store temp matrices, so cleanup is nice